Updated config.go - added defaults config option

JNickson opened the pull request:
Allows for subscription_id and resource_group to have a default value to fall back on per config file for all resources in that file.

JNickson committed:
Updated config.go - added defaults config option

Allows for subscription_id and resource_group to have a default value to fall back on per config file for all resources in that file.

alec-pinson commented:
Haven't looked properly yet because it's hard to read on my phone but might be worth adding an app3.yaml in resources with an example of how to use what you added.

JNickson committed:
Added app3 for example usage

alec-pinson commented on config.go:
```diff
@@ -10,18 +10,29 @@ import (
+}
+
 type ResourceConfiguration struct {
+	Defaults       []string `yaml:"defaults"`
```
Think you added this and don't actually use it ?

alec-pinson commented on config.go:
```diff
@@ -10,18 +10,29 @@ import (
 	Redis       RedisConfiguration      `yaml:"redis"`
 	Auth        Authentication          `yaml:"auth"`
 	Resources   []ResourceConfiguration `yaml:"resources"`
+	Defaults    []Defaults              `yaml:"defaults"`
```
Don't think there's any need for this to be an array of Defaults?

alec-pinson replied:
```suggestion
	Defaults    Defaults              `yaml:"defaults"`
```

alec-pinson commented on config.go:
```diff
@@ -91,71 +104,86 @@ func (c *Configuration) load(reload ...bool) *Configuration {
+		rg := ""
+		for _, resource := range config.Resources {
+			if resource.SubscriptionId == "" {
+				subId = config.Defaults[0].SubscriptionId
```
```suggestion
				subId = config.Defaults.SubscriptionId
```
Don't think this needs to be an array

alec-pinson commented on config.go:
```diff
@@ -91,71 +104,86 @@ func (c *Configuration) load(reload ...bool) *Configuration {
+				subId = resource.SubscriptionId
+			}
+			if resource.ResourceGroup == "" {
+				rg = config.Defaults[0].ResourceGroup
```
```suggestion
				rg = config.Defaults.ResourceGroup
```
Don't think this needs to be an array

alec-pinson commented on config/config.yaml:
```diff
@@ -19,6 +19,10 @@ redis:
   port: 6379
   token: my-sup3r-comp1ic4t3d-s3cr3t-t0k3n
 
+defaults:
+  - subscription_id: notreal-default-subscription
+    resource_group: notreal-default-rg
```
```suggestion
defaults:
  subscription_id: notreal-default-subscription
  resource_group: notreal-default-rg
```
Don't think this needs to be an array/list/slince whatever it is :D

alec-pinson commented on config/resources/app2.yaml:
```diff
@@ -1,3 +1,7 @@
+defaults:
+  - subscription_id: notreal-default-subscription
+    resource_group: notreal-default-rg
```
```suggestion
  subscription_id: notreal-default-subscription
  resource_group: notreal-default-rg
```
Don't think this needs to be an array/list/slince whatever it is :D

alec-pinson commented on config/resources/app3.yaml:
```diff
@@ -0,0 +1,22 @@
+    subscription_id: notreal-not-real-not-notreal
+    resource_group: notreal-rg
+    policy_name: notrealpolicy
+    
```
```suggestion

```
Fix line end

JNickson committed:
removed resource config defaults field in struct

JNickson committed:
Apply suggestions from code review

yeah i thought that it needed to be a slice for some reason but after the change from the struct yeah looks good then lol

Co-authored-by: Alec <30310787+alec-pinson@users.noreply.github.com>

JNickson committed:
fixed whitespace

alec-pinson commented on config.go:
```diff
@@ -91,71 +103,86 @@ func (c *Configuration) load(reload ...bool) *Configuration {
+		for _, resource := range config.Resources {
+			if resource.SubscriptionId == "" {
+				subId = config.Defaults.SubscriptionId
+			} else if resource.SubscriptionId != "" {
```
```suggestion
			} else {
```

alec-pinson commented on config.go:
```diff
@@ -91,71 +103,86 @@ func (c *Configuration) load(reload ...bool) *Configuration {
+			}
+			if resource.ResourceGroup == "" {
+				rg = config.Defaults.ResourceGroup
+			} else if resource.ResourceGroup != "" {
```
```suggestion
			} else {
```

alec-pinson reviewed (changes requested):
Just a lil cleanup on the if else :)

JNickson committed:
Update config.go

Co-authored-by: Alec <30310787+alec-pinson@users.noreply.github.com>

JNickson committed:
changed rg if else aswell
